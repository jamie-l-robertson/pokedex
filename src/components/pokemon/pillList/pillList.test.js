import React from 'react';
import { PillList } from './';

const items = ['Grass', 'Poision'];
const List = mount(<PillList data={items} />);
const ListWithTitle = mount(<PillList data={items} title="Types" />);

describe('<PillList />', () => {
  it('Pill list renders', () => {
    expect(List).toMatchSnapshot();
  });

  it('Has 2 items', () => {
    expect(List.find('li')).toHaveLength(2);
  });

  it('Items are correct', () => {
    expect(List.find('li').get(0).props.children).toEqual('Grass');
  });

  it('Has an optional title', () => {
    expect(ListWithTitle.find('h2').text()).toEqual('Types');
  });
});
