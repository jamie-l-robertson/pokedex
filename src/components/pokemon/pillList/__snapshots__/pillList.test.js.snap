// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PillList /> Pill list renders 1`] = `
<ul>
  <styled.li
    className="pill-grass"
  >
    <li
      className="pill-grass sc-bdVaJa kdfYYr"
    >
      Grass
    </li>
  </styled.li>
  <styled.li
    className="pill-poision"
  >
    <li
      className="pill-poision sc-bdVaJa kdfYYr"
    >
      Poision
    </li>
  </styled.li>
</ul>
`;
